docs/architecture/adrs/0002-use-dto.md

```diff
@@ -0,0 +1,38 @@
+# 1. Record architecture decisions
+
+Date: 2023-09-22
+
+## Status
+
+Pending
+
+## Context
+
+Using a Data Transfer Object (DTO) in the `bichard-next-ui` repo will help separate the User Interface (React) from
+the Database objects. It will also allow to use TypeScript more effectively, enforcing stricter types on objects at
+compile time.
+
+We have chosen the naming convention of:
+
+- Types:
+  - `Display` - Prefix for the object to be consumed in the UI
+  - `DisplayPartial` - Prefix for the object which has limited fields
```

Comment:
We might need to redact different fields in different contexts, for example when showing who left a note we only need the user's name, whereas when looking at who has a case locked we need to also know the user's groups.

Is the intention here to have a single `DisplayPartial` for e.g. a user? I think it would be worth mentioning how we would handle having multiple partial objects for each database object for different contexts, or clarifying the intention to have a single partial object.

Reply:
> We might need to redact different fields in different contexts, for example when showing who left a note we only need the user's name, whereas when looking at who has a case locked we need to also know the user's groups

That has been done for the notes see: https://github.com/ministryofjustice/bichard7-next-ui/blob/main/src/services/dto/noteDto.ts#L17C28-L17C28

> Is the intention here to have a single DisplayPartial for e.g. a user? I think it would be worth mentioning how we would handle having multiple partial objects for each database object for different contexts, or clarifying the intention to have a single partial object.

You can have as many DTOs as you wish. E.g. `DisplayPartialUser`, `DisplayPartialUserForContext` would be a valid use case. Ideally, we want to remove as much data from the Objects so that they only have the bare minimum required to work.